pkg/agent/flowexporter/exporter/exporter.go

```diff
@@ -235,22 +254,26 @@ func (exp *flowExporter) initFlowExporter(collectorAddr string, collectorProto s
 	return nil
 }
 
-func (exp *flowExporter) sendFlowRecords() error {
-	sendAndUpdateFlowRecord := func(key flowexporter.ConnectionKey, record flowexporter.FlowRecord) error {
+func (exp *flowExporter) sendFlowRecord(dataSet ipfix.IPFIXSet) error {
+	addOrUpdateFlowRecord := func(key flowexporter.ConnectionKey, record flowexporter.FlowRecord) error {
 		templateID := exp.templateIDv4
 		if record.IsIPv6 {
 			templateID = exp.templateIDv6
 		}
-		dataSet := ipfix.NewSet(ipfixentities.Data, templateID, false)
-		if err := exp.sendDataSet(dataSet, record); err != nil {
-			return err
+		// if current message size is close to size limit, send the dataSet before adding the new record
+		if dataSet.GetSet().GetBuffLen()+msgHeaderLen > exp.process.GetMsgSizeLimit()*3/4 {
```

Comment:
Curious why 3/4?

Reply:
If it is larger than the limit value, the current set cannot be sent either. So I make it 3/4 just to be safe and under assumption that our current data size is less than 128B (512B/4). Do you have a better suggestion?

Reply:
How about doing this:

`dataSetIPv6.GetSet().GetBuffLen()+msgHeaderLen+recordLen > exp.process.GetMsgSizeLimit() `

recordLen is always the same. We can maintain two global variables recordLenIPv4 and recordLenIPv6 by initializing them to 0.
Fill them up only when we add the first record for IPv4 and IPv6 records, that is when they are 0 in the function `addRecordToSet`.

Reply:
Sounds good. Updated. Thanks!